makefiles/boards/sam0.inc.mk

```diff
@@ -5,15 +5,17 @@ PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 # DEBUG_ADAPTER_ID="ATML..."
 
 # The SERIAL setting is only available for backwards compatibility with older
-# settings.
-ifneq (,$(SERIAL))
-  EDBG_ARGS += --serial $(SERIAL)
-  SERIAL_TTY = $(firstword $(shell $(RIOTTOOLS)/usb-serial/find-tty.sh $(SERIAL)))
-  ifeq (,$(SERIAL_TTY))
-    $(error Did not find a device with serial $(SERIAL))
+# settings. It's only checked when a target is given to the make command.
+ifneq (,$(MAKECMDGOALS))
```

Comment:
This would still fail if `all` or `clean` are given:

```
make -C examples/hello-world BOARD=samr21-xpro SERIAL=ATML211234567 all
make: Entering directory '/home/francisco/workspace/RIOT2/examples/hello-world'
/home/francisco/workspace/RIOT2/makefiles/boards/sam0.inc.mk:14: *** Did not find a device with serial ATML211234567.  Stop.
make: Leaving directory '/home/francisco/workspace/RIOT2/examples/hello-world'
```

Reply:
Ah, yes. Should be fixed now :)